libs/market-depth/src/lib/orderbook-data.ts

```diff
@@ -121,37 +132,38 @@ export const compactRows = (
   resolution: number
 ) => {
   // map raw sell data to OrderbookData
-  const askOrderbookData = [...(sell ?? [])].map<OrderbookRowData>(
+  const askOrderbookData = [...(sell ?? [])].map<PartialOrderbookRowData>(
     mapRawData(VolumeType.ask)
   );
   // map raw buy data to OrderbookData
-  const bidOrderbookData = [...(buy ?? [])].map<OrderbookRowData>(
+  const bidOrderbookData = [...(buy ?? [])].map<PartialOrderbookRowData>(
     mapRawData(VolumeType.bid)
   );
-
   // group by price level
-  const groupedByLevel = groupBy<OrderbookRowData>(
+  const groupedByLevel = groupBy<PartialOrderbookRowData>(
     [...askOrderbookData, ...bidOrderbookData],
     (row) => getPriceLevel(row.price, resolution)
   );
-
-  // create single OrderbookData from grouped OrderbookData[], sum volumes and atore volume by level
-  const orderbookData = Object.keys(groupedByLevel).reduce<OrderbookRowData[]>(
-    (rows, price) =>
-      rows.concat(
-        groupedByLevel[price].reduce<OrderbookRowData>(
-          (a, c) => ({
-            ...a,
-            ask: a.ask + c.ask,
-            askByLevel: Object.assign(a.askByLevel, c.askByLevel),
-            bid: (a.bid ?? 0) + (c.bid ?? 0),
-            bidByLevel: Object.assign(a.bidByLevel, c.bidByLevel),
-          }),
-          createRow(price)
-        )
-      ),
-    []
-  );
+  const orderbookData: OrderbookRowData[] = [];
+  Object.keys(groupedByLevel).forEach((price) => {
+    const row = extendRow(
+      groupedByLevel[price].pop() as PartialOrderbookRowData
+    );
+    row.price = price;
+    let subRow: PartialOrderbookRowData | undefined;
+    // eslint-disable-next-line no-cond-assign
+    while ((subRow = groupedByLevel[price].pop())) {
```

Comment:
nitpick (non-blocking): could we rewrite this to avoid the eslint disable?

```js
let subrow = groupedByLevel[price].pop();
while (subrow) {
  //logic
  subrow = groupedByLevel[price].pop();
}
whle

```